src/main/java/gregtech/api/pipenet/block/BlockPipe.java

```diff
@@ -598,7 +599,8 @@ public boolean hasPipeCollisionChangingItem(IBlockAccess world, BlockPos pos, En
     public boolean hasPipeCollisionChangingItem(IBlockAccess world, BlockPos pos, ItemStack stack) {
         return isPipeTool(stack) || ToolHelper.isTool(stack, ToolClasses.SCREWDRIVER) ||
                 GTUtility.isCoverBehaviorItem(stack, () -> hasCover(getPipeTileEntity(world, pos)),
-                        coverDef -> ICoverable.canPlaceCover(coverDef, getPipeTileEntity(world, pos).getCoverableImplementation()));
+                        coverDef -> getPipeTileEntity(world, pos).getCoverableImplementation()
+                                .canPlaceCoverOnSide(EnumFacing.DOWN)); //TODO figure out dir
```

Comment:
Or let the method pass in null to check if there is any side it can be placed on. Doing raytracing here is not a good idea, since it is used for raytracing.